Sources/Session.php

```diff
@@ -95,151 +98,157 @@ function loadSession()
- *
- * @param string $session_id The session ID
- * @return boolean Whether the session was successfully destroyed
- */
-function sessionDestroy($session_id)
-{
-	global $smcFunc;
+		if (!$this->IsValidSessionID($id))
+			return false;
 
-	if (preg_match('~^[A-Za-z0-9,-]{16,64}$~', $session_id) == 0)
-		return false;
+		// Just delete the row...
+		$smcFunc['db_query']('', '
+			DELETE FROM {db_prefix}sessions
+			WHERE session_id = {string:session_id}',
+			array(
+				'session_id' => $id,
+			)
+		);
 
-	// Just delete the row...
-	$smcFunc['db_query']('', '
-		DELETE FROM {db_prefix}sessions
-		WHERE session_id = {string:session_id}',
-		array(
-			'session_id' => $session_id,
-		)
-	);
+		return true;
+	}
 
-	return true;
-}
+	/**
+	 * Implementation of SessionHandler::GC() replacing the standard gc handler.
+	 * Callback for garbage collection.
+	 *
+	 * @param int $max_lifetime The maximum lifetime (in seconds) - prevents deleting of sessions older than this
+	 * @return boolean Whether the option was successful
+	 */
+	#[\ReturnTypeWillChange]
+	public function gc(/*PHP 8.0 int*/$max_lifetime)/*PHP 8.1 : int|false*/
+	{
+		global $modSettings, $smcFunc;
+
+		// Just set to the default or lower?  Ignore it for a higher value. (hopefully)
+		if (!empty($modSettings['databaseSession_lifetime']) && ($max_lifetime <= 1440 || $modSettings['databaseSession_lifetime'] > $max_lifetime))
+			$max_lifetime = max($modSettings['databaseSession_lifetime'], 60);
+
+		// Clean up after yerself ;).
+		$session_update = $smcFunc['db_query']('', '
+			DELETE FROM {db_prefix}sessions
+			WHERE last_update < {int:last_update}',
+			array(
+				'last_update' => time() - $max_lifetime,
+			)
+		);
+
+		return ($smcFunc['db_affected_rows']() == 0 ? false : true);
+	}
 
-/**
- * Implementation of sessionGC() replacing the standard gc handler.
- * Callback for garbage collection.
- *
- * @param int $max_lifetime The maximum lifetime (in seconds) - prevents deleting of sessions older than this
- * @return boolean Whether the option was successful
- */
-function sessionGC($max_lifetime)
-{
-	global $modSettings, $smcFunc;
-
-	// Just set to the default or lower?  Ignore it for a higher value. (hopefully)
-	if (!empty($modSettings['databaseSession_lifetime']) && ($max_lifetime <= 1440 || $modSettings['databaseSession_lifetime'] > $max_lifetime))
-		$max_lifetime = max($modSettings['databaseSession_lifetime'], 60);
-
-	// Clean up after yerself ;).
-	$session_update = $smcFunc['db_query']('', '
-		DELETE FROM {db_prefix}sessions
-		WHERE last_update < {int:last_update}',
-		array(
-			'last_update' => time() - $max_lifetime,
-		)
-	);
-
-	return ($smcFunc['db_affected_rows']() == 0 ? false : true);
+	/**
+	 * Validates a given string conforms to our testing for a valid session id.
+	 *
+	 * @param string $id The session ID
+	 * @return boolean Whether the string is valid format or not
+	 */
+	private function IsValidSessionID(string $id): bool
```

Comment:
Why not `isValidSessionID`? It's just a method, not a class.

Reply:
Fixed